beater/beater.go

```diff
@@ -45,19 +43,17 @@ func (bt *ApmServer) Run(b *beat.Beat) error {
 		go bt.client.PublishAll(events)
 	}
 
-	bt.server, err = server.New(bt.config.Server)
-	if err != nil {
-		return err
-	}
-	bt.server.Start(callback, bt.config.Host)
-	defer bt.server.Stop()
+	server := apmServer(bt.config, callback)
+	start(server, bt.config.SSL)
```

Comment:
I see the advantage of how you refactored the config passing. For creating, starting and stopping I would be in favor of having this as methods of the server though. What do you think about:
`type apmServer struct { server *http.Server }`

and changing:
`func apmserver(config Config, publish successCallback) *http.Server` -> `func create(config Config, publish successCallback) apmServer`
`func start(server *http.Server, ssl *SSLConfig) `-> `func (*apmServer) start(ssl *SSLConfig) `
`func stop(server *http.Server) error`-> `func (*apmServer) stop() error`